.github/workflows/php-test-plugins.yml

```diff
@@ -68,6 +68,20 @@ jobs:
       - name: Composer Install
         run: npm run wp-env run tests-cli -- --env-cwd="wp-content/plugins/$(basename $(pwd))" composer install --no-interaction --no-progress
       - name: Running single site unit tests
-        run: npm run test-php
+        run: npm run test-php -- --coverage-clover=coverage.xml
```

Comment:
Since collecting code coverage takes much longer, as I recall, should the coverage only happen for one of the jobs instead of for all of them?

Reply:
Yes you're right, collecting code coverage takes time, but I'm not quite sure that for which job it'll be ideal to run.
Should I run for `PHP v8.0` and if this is a viable option, then will the below implementation work since we are using the latest WP version anyways?
```yml

      - name: Running single site unit tests with coverage
        if: matrix.php == '8.0'
        run: npm run test-php -- --coverage-clover=coverage.xml
      - name: Running multisite unit tests with coverage
        if: matrix.php == '8.0'
        run: npm run test-php-multisite -- --coverage-clover=coverage-multisite.xml

      - name: Upload single site coverage reports to Codecov
        if: matrix.php == '8.0'
        uses: codecov/codecov-action@v4
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          files: coverage.xml
          flags: unittests
          name: ${{ matrix.php }}-single-site-coverage
      - name: Upload multisite coverage reports to Codecov
        if: matrix.php == '8.0'
        uses: codecov/codecov-action@v2
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          files: coverage-multisite.xml
          flags: multisite
          name: ${{ matrix.php }}-multisite-coverage
```

cc : @westonruter

Reply:
You may want to refer to what was done for the AMP plugin: https://github.com/ampproject/amp-wp/blob/60a655cdad9e3431b47b2dcc3b2e5ddb86ac5c5e/.github/workflows/build-test-measure.yml#L261-L337

Note the new `coverage` matrix variable which is only set to `true` for one configuration. This would be better than having to test against a specific PHP version everywhere. Yes, I think running coverage on the most recent PHP version would be best as it should complete the fastest. So maybe do it in PHP 8.2?

cc @thelovekesh